components/message_identifier_router.jsx

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {joinChannel} from 'mattermost-redux/actions/channels';
+import {getUser, getUserByUsername, getUserByEmail} from 'mattermost-redux/actions/users';
+
+import ChannelView from 'components/channel_view';
+import UserStore from 'stores/user_store.jsx';
+import ChannelStore from 'stores/channel_store.jsx';
+import TeamStore from 'stores/team_store.jsx';
+import {Constants} from 'utils/constants.jsx';
+import {openDirectChannelToUser} from 'actions/channel_actions.jsx';
+import * as GlobalActions from 'actions/global_actions.jsx';
+import * as Utils from 'utils/utils.jsx';
+import store from 'stores/redux_store.jsx';
+const dispatch = store.dispatch;
+const getState = store.getState;
+
+const LengthOfId = 26;
+const LengthOfGroupId = 40;
+const LengthOfUserIdPair = 54;
+
+function onChannelByIdentifierEnter({match, history}) {
+    const {path, identifier} = match.params;
+
+    if (path === 'channels') {
+        if (identifier.length === LengthOfId) {
```

Comment:
What happens if I have a channel name that happens to be 26 characters long? Is it going to try it as a channel ID and fail?

Reply:
I've found I cannot enter more than 22 characters for a channel name in the webapp. if it could be in another way and that has not same length as specified, it will going to try it as a channel name, and fail (replaced to fallback URL).